local-cluster/tests/local_cluster.rs

```diff
@@ -5706,6 +5703,7 @@ fn test_randomly_mixed_block_verification_methods_between_bootstrap_and_not() {
 
 /// Forks previous marked invalid should be marked as such in fork choice on restart
 #[test]
+#[ignore]
 #[serial]
```

Comment:
yeah, but looks like if I make either of patches mentioned in https://github.com/anza-xyz/agave/pull/823/files#r1575382916 it passes. so looks like it relies on contact-infos propagation by a certain time which is not guaranteed anyways.

Nonetheless, we are not going to backport this patch, and as soon as mainnet has upgraded to https://github.com/anza-xyz/agave/pull/803 we can use the new contact-info in the pull requests which makes the test pass.

But agreed that we should investigate this test and figure out what assumptions are failing. Ashwin has opened https://github.com/anza-xyz/agave/issues/982 to keep track of this.

Reply:
ya good point. the frequency fix is interesting. Wondering why that affects the ContactInfo but not LegacyContactInfo? I'd think it wouldn't have an affect...

Reply:
> the frequency fix is interesting. Wondering why that affects the ContactInfo but not LegacyContactInfo? 

Previously, in addition to push and pull-responses, one more way that contact-infos were propagated across cluster was through the "caller" in pull-requests:
https://github.com/anza-xyz/agave/blob/e7452b768/gossip/src/cluster_info.rs#L2076-L2081
https://github.com/anza-xyz/agave/blob/e7452b768/gossip/src/crds_gossip_pull.rs#L296-L307
`process_pull_requests` is terribly named but basically what it does it upserts CRDS table with the caller's contact-info attached in pull-requests.

With this commit we are changing what struct counts as "contact-info" by switching from `LegacyContactInfo` to `ContactInfo`. So that caller's (legacy) contact-info attached to pull-requests is no longer relevant (because it is not a `ContactInfo`, but a `Legacy...`).

So basically with this commit we are killing that extra propagation path for contact-infos I mentioned earlier above. Increasing the frequency of "push_self" seems to compensate for that.

Reply:
ahhhh yes got it. that makes sense